### ISSUE TITLE
[Teams][Rendering] Markdown formatted text containing images not rendering on Teams

### ISSUE THREAD

**Issue by YJasonY** [microsoft/AdaptiveCards#4121]
# Platform

Microsoft Teams

# Version of SDK

1.2

# Details

We have a bot that is sending adaptive card responses on Teams. The bot renders results from QnA Maker answers into a card. The QnA Maker answers are formatted using markdown and some answers contain images. These images are not showing up on Teams, but do show up in the designer and the Bot Emulator.

The issue seems to exist in the [FAQ Plus template](https://github.com/OfficeDev/microsoft-teams-faqplusplus-app) as well.

## QnA Answer (Markdown)

```
Click **Teams**\n\n![Teams button](https://support.content.office.net/en-us/media/20305b5c-df47-4154-a21a-b12712c4224c.png)\n\non the left side of the app, then click **Add team** at the bottom of your teams list. Go to the first tile in the **Suggested teams** view and click **Create a team**. From there, you can invite people or even entire contact groups (aka "distribution lists") to join your team. Now that you have a new team, why not [add some new channels](https://support.office.com/en-us/article/create-a-channel-in-teams-fda0b75e-5b90-4fb8-8857-7e102b014525) for specific conversation topics? Or even [invite guests](https://support.office.com/en-us/article/add-guests-to-a-team-in-teams-fccb4fa6-f864-4508-bdde-256e7384a14f) from outside your organization to help with a project?
```

The content was generated by QnA Maker from the following [Teams FAQ page](https://support.microsoft.com/en-us/office/faq-f4644010-d5fa-4055-b42a-6a5317316e18?ui=en-us&rs=en-us&ad=us)

## JSON

```json
{
    "type": "AdaptiveCard",
    "version": "1.2",
    "body": [
        {
            "type": "TextBlock",
            "size": "Medium",
            "weight": "Bolder",
            "text": "'Click **Teams** ![Teams button](https://support.content.office.net/en-us/media/20305b5c-df47-4154-a21a-b12712c4224c.png) on the left side of the app, then click **Add team** at the bottom of your teams list. Go to the first tile in the **Suggested teams** view and click **Create a team**. From there, you can invite people or even entire contact groups (aka \"distribution lists\") to join your team. Now that you have a new team, why not [add some new channels]([https://support.office.com/en-us/article/create-a-channel-in-teams-fda0b75e-5b90-4fb8-8857-7e102b014525](https://support.office.com/en-us/article/create-a-channel-in-teams-fda0b75e-5b90-4fb8-8857-7e102b014525)) for specific conversation topics? Or even [invite guests]([https://support.office.com/en-us/article/add-guests-to-a-team-in-teams-fccb4fa6-f864-4508-bdde-256e7384a14f](https://support.office.com/en-us/article/add-guests-to-a-team-in-teams-fccb4fa6-f864-4508-bdde-256e7384a14f)) from outside your organization to help with a project?'",
            "wrap": true
        },
        {
            "type": "TextBlock",
            "text": "I think that is the answer (confidence score of 95%)",
            "wrap": true
        }
    ],
    "actions": [
        {
            "type": "Action.Submit",
            "data": {
                "type": "BotHelpful",
                "answer": "Click **Teams** ![Teams button](https://support.content.office.net/en-us/media/20305b5c-df47-4154-a21a-b12712c4224c.png) on the left side of the app, then click **Add team** at the bottom of your teams list. Go to the first tile in the **Suggested teams** view and click **Create a team**. From there, you can invite people or even entire contact groups (aka \"distribution lists\") to join your team. Now that you have a new team, why not [add some new channels](https://support.office.com/en-us/article/create-a-channel-in-teams-fda0b75e-5b90-4fb8-8857-7e102b014525) for specific conversation topics? Or even [invite guests](https://support.office.com/en-us/article/add-guests-to-a-team-in-teams-fccb4fa6-f864-4508-bdde-256e7384a14f) from outside your organization to help with a project?",
                "questionId": 158,
                "qnaId": 104
            },
            "title": "This is helpful"
        },
        {
            "type": "Action.Submit",
            "data": {
                "type": "BotUnhelpful",
                "questionId": 158,
                "qnaId": 0,
                "upn": "user@email.com"
            },
            "title": "Tag Expert"
        }
    ],
    "$schema": "http://adaptivecards.io/schemas/adaptive-card.json"
}
```

## Web Chat Designer ✅
![image](https://user-images.githubusercontent.com/16281323/83822939-d94bd980-a715-11ea-8a4c-daeb9dc8c24b.png)

## Microsoft Teams Designer ✅
![image](https://user-images.githubusercontent.com/16281323/83823367-35632d80-a717-11ea-93a2-19c5be5885a1.png)

## Microsoft Teams ❌
![image](https://user-images.githubusercontent.com/16281323/83823064-40698e00-a716-11ea-8158-37af0b9b231c.png)

**Comment by shalinijoshi19** [comment]
@siduppal I believe this may be a Teams issue- Could you help take a look please? 
I checked on our designer and it seems to render correctly here: 
![image](https://user-images.githubusercontent.com/20822397/83835408-93572d00-a6a5-11ea-8f6f-e9c9853a1d0f.png)

**Comment by pushyami-consultesoft** [comment]
@YJasonY did this issue resolved. if yes how? I am also facing the same issue

**Comment by JeSuisUnCaillou** [comment]
Also facing the issue, images do not render in markdown format

**Comment by JeSuisUnCaillou** [comment]
I just found a workaround, and possibly an explanation of the bug.

This is the url of my image, in the markdown I send in my markdown : 

> In my markdown
> `![image](https://canbot-jam.s3.amazonaws.com/2021-02-10/1612972263-capture_d___e__cran_2021-02-10_a___16.49.06.png)`

When I inspect my image in the teams web app and find its `src`, I can see it in the url param of a longer url, on skype.com :

> This image doesn't work
> https://urlp.asm.skype.com/v1/url/content?url=https%3a%2f%2fcanbot-jam.s3.amazonaws.com%2f2021-02-10%2f1612972263-capture_d___e__cran_2021-02-10_a___16.49.06.png

I imagine this is an asset hosting service, which hosts the image redirects to the hosted url. And I can see that this image is 401 forbidden in the console.

After a thousand tests, I ended up having a random image displayed, and when I inspected it I saw a different hosting domain in the `src` :

> This image works
> https://fr-prod.asyncgw.teams.microsoft.com/urlp/v1/url/content?url=https%3a%2f%2fparticuliers.engie.fr%2fcontent%2fdam%2fparticuliers-engie%2fphotos%2fhp%2fok-illu-aucun-parrainage.png

So I tried to force the same hosting domain by putting it myself before my image url :

> In my markdown
> `![image](https://fr-prod.asyncgw.teams.microsoft.com/urlp/v1/url/content?url=https%3a%2f%2fcanbot-jam.s3.amazonaws.com%2f2021-02-10%2f1612972263-capture_d___e__cran_2021-02-10_a___16.49.06.png)`

And it worked !

I guess `urlp.asm.skype.com` currently have some issues.

TL;DR : prefix your image url with `https://fr-prod.asyncgw.teams.microsoft.com/urlp/v1/url/content?url=`
(and encode the image url to be valid in the param, ofc)

**Comment by xu1718191411** [comment]
> I just found a workaround, and possibly an explanation of the bug.
> 
> This is the url of my image, in the markdown I send in my markdown :
> 
> > In my markdown
> > `![image](https://canbot-jam.s3.amazonaws.com/2021-02-10/1612972263-capture_d___e__cran_2021-02-10_a___16.49.06.png)`
> 
> When I inspect my image in the teams web app and find its `src`, I can see it in the url param of a longer url, on skype.com :
> 
> > This image doesn't work
> > https://urlp.asm.skype.com/v1/url/content?url=https%3a%2f%2fcanbot-jam.s3.amazonaws.com%2f2021-02-10%2f1612972263-capture_d___e__cran_2021-02-10_a___16.49.06.png
> 
> I imagine this is an asset hosting service, which hosts the image redirects to the hosted url. And I can see that this image is 401 forbidden in the console.
> 
> After a thousand tests, I ended up having a random image displayed, and when I inspected it I saw a different hosting domain in the `src` :
> 
> > This image works
> > https://fr-prod.asyncgw.teams.microsoft.com/urlp/v1/url/content?url=https%3a%2f%2fparticuliers.engie.fr%2fcontent%2fdam%2fparticuliers-engie%2fphotos%2fhp%2fok-illu-aucun-parrainage.png
> 
> So I tried to force the same hosting domain by putting it myself before my image url :
> 
> > In my markdown
> > `![image](https://fr-prod.asyncgw.teams.microsoft.com/urlp/v1/url/content?url=https%3a%2f%2fcanbot-jam.s3.amazonaws.com%2f2021-02-10%2f1612972263-capture_d___e__cran_2021-02-10_a___16.49.06.png)`
> 
> And it worked !
> 
> I guess `urlp.asm.skype.com` currently has some issues.
> 
> TL;DR : prefix your image url with `https://fr-prod.asyncgw.teams.microsoft.com/urlp/v1/url/content?url=`
> (and encode the image url to be valid in the param, ofc)

That is a really good trick

**Comment by mpelekh** [comment]
The issue is still reproduced. Any updates on it?

> TL;DR : prefix your image url with https://fr-prod.asyncgw.teams.microsoft.com/urlp/v1/url/content?url=
(and encode the image url to be valid in the param, ofc)

The suggested foregoing workaround doesn't work.

**Comment by Prasad-MSFT** [comment]
@mpelekh , we are able to repro the issue and raised a bug for this. Concerned team is looking into it. We will inform you once we get any update.
Thanks!

**Comment by Wajeed-msft** [comment]
This issue has been copied over to Microsoft Teams repo here: https://github.com/MicrosoftDocs/msteams-docs/issues/4808
